src/kit/Theme/UI.tsx

```diff
@@ -1,101 +1,129 @@
+    .map((char, index) => {
+      return char === char.toUpperCase() ? `${index !== 0 ? '-' : ''}${char.toLowerCase()}` : char;
+    })
+    .join('');
 };
 
-const StoreActionUI = {
-  HideUIChrome: 'HideUIChrome',
-  HideUISpinner: 'HideUISpinner',
-  SetMode: 'SetMode',
-  SetPageVisibility: 'SetPageVisibility',
-  SetTheme: 'SetTheme',
-  ShowUIChrome: 'ShowUIChrome',
-  ShowUISpinner: 'ShowUISpinner',
-} as const;
+interface ThemeSettings {
+  className: string;
+  themeIsDark: boolean;
+  theme: Theme;
+}
 
-type ActionUI =
-  | { type: typeof StoreActionUI.HideUIChrome }
-  | { type: typeof StoreActionUI.HideUISpinner }
-  | { type: typeof StoreActionUI.SetMode; value: Mode }
-  | { type: typeof StoreActionUI.SetPageVisibility; value: boolean }
-  | { type: typeof StoreActionUI.SetTheme; value: { darkLight: DarkLight; theme: Theme } }
-  | { type: typeof StoreActionUI.ShowUIChrome }
-  | { type: typeof StoreActionUI.ShowUISpinner };
+const UIContext = React.createContext<ThemeSettings | undefined>(undefined);
 
-class UIActions {
-  constructor(private dispatch: Dispatch<ActionUI>) {}
+export const useTheme = (): { themeSettings: ThemeSettings, getThemeVar: (name: ThemeVariable) => string; } => {
+  /**
+   * Some UI Kit components such as the CodeEditor do not inherit the theme from css or page styling
+   * and instead require us to set a theme related prop dynamically. This context allows us to
+   * subscribe to UIProvider theme updates and re-render these child components with the correct
+   * theme.
+   */
+  const context = useContext(UIContext);
+  if (context === undefined) {
+    throw new Error('useStore(UI) must be used within a UIProvider');
+  }
 
-  public hideChrome = (): void => {
-    this.dispatch({ type: StoreActionUI.HideUIChrome });
-  };
+  const getThemeVar = (name: ThemeVariable): string => {
+    return context.theme[name]
+  }
 
-  public hideSpinner = (): void => {
-    this.dispatch({ type: StoreActionUI.HideUISpinner });
-  };
+  return { themeSettings: context, getThemeVar };
+};
 
-  public setMode = (mode: Mode): void => {
-    this.dispatch({ type: StoreActionUI.SetMode, value: mode });
-  };
 
-  public setPageVisibility = (isPageHidden: boolean): void => {
-    this.dispatch({ type: StoreActionUI.SetPageVisibility, value: isPageHidden });
-  };
+export const UIProvider: React.FC<{
+  children?: React.ReactNode;
+  themeIsDark?: boolean;
+  theme: Theme;
+}> = ({ children, theme, themeIsDark = false }) => {
+  const className = `ui-provider-${Math.random().toString(36).substring(2, 9)}`;
 
-  public setTheme = (darkLight: DarkLight, theme: Theme): void => {
-    this.dispatch({ type: StoreActionUI.SetTheme, value: { darkLight, theme } });
-  };
+  useEffect(() => {
+    const styles: string[] = [];
+    Object.keys(globalCssVars).forEach((key) => {
+      const value = (globalCssVars as Record<RecordKey, string>)[key];
+      if (value) {
+        styles.push(`--${camelCaseToKebab(key)}:${value}`);
```

Comment:
is the push here required if you're going to add the same style to the document element?